conseil-lorre/src/main/scala/tech/cryptonomic/conseil/indexer/bitcoin/BitcoinIndexer.scala

```diff
@@ -0,0 +1,128 @@
+package tech.cryptonomic.conseil.indexer.bitcoin
+
+import java.util.concurrent.Executors
+
+import scala.concurrent.{ExecutionContext, Future}
+
+import cats.effect.{IO, Resource}
+import com.typesafe.scalalogging.LazyLogging
+import org.http4s.headers.Authorization
+import org.http4s.BasicCredentials
+import org.http4s.client.blaze.BlazeClientBuilder
+import slick.jdbc.PostgresProfile.api._
+import slickeffect.Transactor
+import slickeffect.transactor.{config => transactorConfig}
+
+import tech.cryptonomic.conseil.common.util.DatabaseUtil
+import tech.cryptonomic.conseil.indexer.config.LorreConfiguration
+import tech.cryptonomic.conseil.common.config.Platforms
+import tech.cryptonomic.conseil.common.config.Platforms.BitcoinConfiguration
+import tech.cryptonomic.conseil.indexer.LorreIndexer
+import tech.cryptonomic.conseil.indexer.logging.LorreProgressLogging
+import tech.cryptonomic.conseil.common.rpc.RpcClient
+
+/**
+  * Class responsible for indexing data for Bitcoin Blockchain.
+  *
+  * @param lorreConf Lorre configuration
+  * @param bitcoinConf Bitcoin configuration
+  */
+class BitcoinIndexer(
+    lorreConf: LorreConfiguration,
+    bitcoinConf: BitcoinConfiguration
+) extends LazyLogging
+    with LorreIndexer
+    with LorreProgressLogging {
+
+  /**
+    * Executor for the rpc client, timer and to handle stop method.
+    */
+  private val indexerExecutor = Executors.newFixedThreadPool(bitcoinConf.batchingConf.indexerThreadsCount)
+
+  /**
+    * Dedicated executor for the http4s.
+    */
+  private val httpExecutor = Executors.newFixedThreadPool(bitcoinConf.batchingConf.httpFetchThreadsCount)
+
+  /**
+    * [[cats.ContextShift]] is the equivalent to [[ExecutionContext]], 
+    * it's used by the Cats Effect related methods.
+    */
+  implicit val contextShift = IO.contextShift(ExecutionContext.fromExecutor(indexerExecutor))
```

Comment:
Can we make private, what should be private? I guess here, there is only one method that should be public (or two, don't remember). I really like to keep it like that, because that can speed up future refactoring and tells instantly which method is actually used or not. Let's think about classes, changes in scopes, not in modules :)

Reply:
Please review the code again